### PR TITLE
switch config ui to backend and add notifs

### PR DESCRIPTION
**Pull request by AbdelHedhili**
Signed-off-by: AbdelHedhili <abdelsalem.hedhili@rte-france.com>

### CODE REVIEW

**Review by ChamseddineBhd — APPROVED** [review]

**Review by geofjamg — CHANGES_REQUESTED** [review]
As discused last week in standup meeting: theme is a very special case because we need to get the theme before being logged in and we cannot get parameter frol server before being logged. So for the theme we are going to use both server side saving and local storage so that when we load the application, we first use local storage param to get the theme, then once we are logged we get one from server.
When we change the theme in the UI OR when the theme is changed by a notification, we also update the local storage.

File: src/redux/reducer.js [hunk 1]
```diff
@@ -75,21 +51,21 @@ const initialState = {
...
+    useName: true,
     user: null,
-    centerLabel: getLocalStorageCenterLabel(),
-    diagonalLabel: getLocalStorageDiagonalLabel(),
-    lineFullPath: getLocalStorageLineFullPath(),
-    lineParallelPath: getLocalStorageLineParallelPath(),
-    lineFlowMode: getLocalStorageLineFlowMode(),
-    lineFlowColorMode: getLocalStorageLineFlowColorMode(),
-    lineFlowAlertThreshold: getLocalStorageLineFlowAlertThreshold(),
+    centerLabel: false,
```
Comment (SlimaneAmar) [inline src/redux/reducer.js:56]: It is not possible to use a dictionary ?

Comment (AbdelHedhili) [reply src/redux/reducer.js:56]: what do you mean ?

File: src/components/parameters.js [hunk 2]
```diff
@@ -125,36 +123,59 @@ const Parameters = ({ showParameters, hideParameters }) => {
 
     const handleChangeTheme = (event) => {
         const theme = event.target.value;
-        dispatch(selectTheme(theme));
+        let json = JSON.stringify([
```
Comment (geofjamg) [inline src/components/parameters.js:126]: Why do you strinify the object here ? This is really ugly in the code and cumbersome. If you need to stringify just to put it in the body (not sure this is necessaey, to check) why not doing it in the  updateConfigParameters function?

File: src/components/app.js [hunk 3]
```diff
@@ -206,6 +309,25 @@ const App = () => {
         }
     }, [user]);
 
+    useEffect(() => {
+        if (user !== null) {
+            fetchConfigParameters().then((params) => {
+                console.debug('received UI parameters :');
+                console.debug(params);
+                //if it's the user first connexion we want to set the default parameters in the database
```
Comment (geofjamg) [inline src/components/app.js:317]: Why ?

**Review by geofjamg — CHANGES_REQUESTED** [review]

**Review by geofjamg — APPROVED** [review]

**Pull request merged**